src/gui/banlistoptionsdialog.cpp

```diff
@@ -107,7 +107,9 @@ void BanListOptionsDialog::on_buttonBanIP_clicked()
 
 void BanListOptionsDialog::on_buttonDeleteIP_clicked()
 {
-    const QModelIndexList selection = m_ui->bannedIPList->selectionModel()->selectedIndexes();
+    QModelIndexList selection = m_ui->bannedIPList->selectionModel()->selectedIndexes();
+    //sort indices in descent order to prevent shift of undeleted entries while deleting
```

Comment:
@dzmat:  Minor grammar issue in this comment. 

> //sort indices in **descent** order 

…should probably be… 

> // sort indices in **descending** order

Reply:
fixed.